media-libs/libextractor/libextractor-1.3.recipe

```diff
@@ -0,0 +1,102 @@
+SUMMARY="Meta data extraction tool and library"
+DESCRIPTION="GNU libextractor is a library used to extract meta data from files of \
+arbitrary type.  It is designed to use helper-libraries to perform the \
+actual extraction, and to be trivially extendable by linking against \
+external extractors for additional file types.  Common use-cases for \
+GNU libextractor include detail-views in file managers, detailed \
+search results in file-sharing networks and general information \
+gathering in forensics investigations and penetration testing."
+HOMEPAGE="https://www.gnu.org/software/libextractor/"
+COPYRIGHT="2013 GNU Project"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="ftp://ftp.gnu.org/gnu/libextractor/libextractor-1.3.tar.gz"
+CHECKSUM_SHA256="868ad64c9a056d6b923d451d746935bffb1ddf5d89c3eb4f67d786001a3f7b7f"
+PATCHES="libextractor-1.3.patchset"
```

Comment:
You can make the patch GCC2-only; e.g. https://github.com/haikuports/haikuports/blob/75a3cf5384f650242f4b8a1094ab305d33ed99e7/dev-vcs/bzr/bzr-2.6.0.recipe#L14